src/reporters/playwright.js

```diff
@@ -105,12 +107,30 @@ export default class Reporter {
 			detail.setBrowser(browser);
 		}
 
-		if (status === 'passed') {
-			detail.setPassed({ override: true });
-		} else if (status === 'skipped') {
-			detail.setSkipped({ override: true });
+		if (expectedStatus === 'passed') {
+			if (status === 'passed') {
+				detail.setPassed({ override: overrideStatus });
+			} else if (status === 'skipped') {
+				detail.setSkipped({ override: overrideStatus });
+			} else {
+				detail.setFailed({ override: overrideStatus });
+			}
+		} else if (expectedStatus === 'failed') {
+			if (status === 'failed') {
+				detail.setPassed({ override: overrideStatus });
+			} else if (status === 'skipped') {
+				detail.setSkipped({ override: overrideStatus });
+			} else {
+				detail.setFailed({ override: overrideStatus });
+			}
+		} else if (expectedStatus === 'skipped') {
+			if (status === 'skipped') {
+				detail.setSkipped({ override: overrideStatus });
+			} else {
+				detail.setFailed({ override: overrideStatus });
+			}
 		} else {
-			detail.setFailed({ override: true });
+			detail.setFailed({ override: overrideStatus });
 		}
```

Comment:
This is the main rework. Much more status checking.

Reply:
It can probably be simplified but I'm just not taking any chances. This covers all the option simply and plainly. I can look at optimizing it later.

Reply:
Yes, was just thinking it'll be much easier now with tests in place! 🙌

Reply:
Fair, I can do the change in another PR. Just want to get this in fast so I can get all the downstreams generating the correct data.